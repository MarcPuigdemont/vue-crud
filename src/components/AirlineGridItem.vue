<template>
  <b-card :title="airline.name" :sub-title="airline.iata" class="airline-card" data-test="airline-card" @click="$emit('edit', airline)">
    <b-row no-gutters>
      <div :style="{ 'background-color': airline.primary_color }" class="airline-card__color w-50" />
      <div :style="{ 'background-color': airline.secondary_color }" class="airline-card__color w-50" />
    </b-row>
    <AirlineListItemServices :services="servicesList" />
  </b-card>
</template>

<script>
import { BCard, BRow } from 'bootstrap-vue';
import AirlineListItemServices from './AirlineListItemServices.vue';

export default {
  name: 'AirlineGridItem',
  props: {
    airline: Object,
  },
  components: {
    AirlineListItemServices,
    BCard,
    BRow,
  },
  computed: {
    servicesList: function() {
      return Object.keys(this.airline.services).map(service => ({ name: service, available: this.airline.services[service] }));
    },
  },
};
</script>
<style scoped>
.airline-card {
  cursor: pointer;
  margin-bottom: 10px;
}
.airline-card__color {
  height: 20px;
  margin-bottom: 10px;
}
</style>
