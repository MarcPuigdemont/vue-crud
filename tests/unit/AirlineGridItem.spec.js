import { mount } from '@vue/test-utils';
import AirlineGridItem from '@/components/AirlineGridItem.vue';

describe('AirlineGridItem', () => {
  test('should match the snapshot', () => {
    const propsData = {
      airline: {
        iata: 'BX',
        name: 'Airline name',
        primary_color: '#ffffff',
        secondary_color: '#000000',
        services: { bags: true, checkin: false, seats: true },
      },
    };
    const wrapper = mount(AirlineGridItem, {
      propsData,
      stubs: {
        AirlineListItemServices: '<div class="airline-list-item-services"></div>',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
    const expectedServicesList = [{ name: 'bags', available: true }, { name: 'checkin', available: false }, { name: 'seats', available: true }];
    expect(wrapper.vm.servicesList).toEqual(expectedServicesList);
  });
});
