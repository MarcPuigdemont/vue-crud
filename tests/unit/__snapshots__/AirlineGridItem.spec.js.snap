// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirlineGridItem should match the snapshot 1`] = `
<div
  class="card airline-card"
  data-test="airline-card"
>
  <!---->
  <!---->
  <div
    class="card-body"
  >
    <h4
      class="card-title"
    >
      Airline name
    </h4>
    <h6
      class="card-subtitle text-muted mb-2"
    >
      BX
    </h6>
    <div
      class="row no-gutters"
    >
      <div
        class="airline-card__color w-50"
        style="background-color: rgb(255, 255, 255);"
      />
       
      <div
        class="airline-card__color w-50"
        style="background-color: rgb(0, 0, 0);"
      />
    </div>
     
    <div
      class="airline-list-item-services"
    />
  </div>
  <!---->
  <!---->
</div>
`;
